doximity/odbc_adapter
# Pull lots of fixes and improvements from Springbuk's fork

- Tests are passing in Docker container
- Reenable Rubocop, add encoding config parameter (#29)
- Handle connection failure by trying to reconnect and translating the error to ConnectionFailedError so the caller knows to retry
- Bump version
- Change gemspecs and gemfile
- Upgrade to rails5.2.1
- Upgrade to rails 6
- Backward compatibility
- Update bind params
- Change params for Column class initializer
- Support json type field
- Support json and date datatype
- Typo fix
- Multi-database support
- [APP-2051] Update Column#initialize to pass on **kwargs (rails 6.1 compatibility)
- Update schema_statements.rb
- Corrections for handling names
- Oops
- Changes because arg list to ActiveRecord::ConnectionAdapters::Column.new changed
- Change to return primary key name as lowercase
- Changes for columns function to return better types
- [APP-2295] Update signature of translate_exception for ruby 3
- Correct column default values
- Add better return types
- Clean up dev code
- Add better handling of other types of null
- Claims page was broken, discovered the bug is in the ODBC Adapter. dbms_type_cast needs to return rows at the end. Also added code to raise exceptions, since currently these circumstances shouldn't be encountered and if that changes spontaneously (such as through an snowflake odbc driver update) we want to know rather than silently have nil values.
- Changes to eliminate garbage collection warning
- MOJ-137 Can't quote error is caused by active_record not knowing what types columns are Updated infrastructure needed for models to know what types their attributes are so it will properly handle type casting automatically. This also removes the necessity of specifying json attributes on models. The only reason that dbms_type_cast is still useful is for when we're executing raw queries instead of using the object model.
- MOJ-150 MOJ-151 Snowflake type mapping & array_of... types for Adds mapping for the remaining snowflake types, removes mapping for JSON type (as this isn't actually a type returned by snowflake ever) Add array_of_intgers, array_of_binaries, etc... types that can be used with attribute tagging to quickly perform useful type coercion.
- MOJ-154 This is all of the work to get objects and arrays quoted properly for the format we'll need them in for updates. Next task is to control saves so that they omit objects/arrays on insert and then go back to update them after the fact
- MOJ-156 This is a working methodology for automatically generating IDs. EasyIdentified can be safely included in the ApplicationRecord as it will do nothing under most circumstances. Same for InsertAttributeStripper Still need to do a little bit of cleanup, but wanted to get the code committed and pushed in it's current state.
- MOJ-156 Removing the AutoIdentified concern. No longer necessary or useful after reworking the EasyIdentified concern.
- MOJ-156 updates from comments Updated striped to stripped Added snowflake integer to the typemap and went ahead and updated all of the new types that were added to be specific to the odbc adapter now that I've discovered that's a thing.
- MOJ-178 Updates to issues identified when working on MOJ-178 EasyIdentified now uses the connection of it's own class instead of using the connection of ApplicationRecord. - The connection of ApplicationRecord in springbuk is the postgres connection InsertAttributeStripped was bugged. Inserts worked fine, updates did not. Added necessary bits to fix that and realized that save and save! had grown to a considerable size with the only difference between them being the method that they called. Refactored them to call a common function and pass a method into that function so it calls the correct parent method.
- MOJ-178 More fixes discovered while working on MOJ-178 RecordNotUnique, QueryTimeoutError, and ConnectionFailedError all inherit from StatementInvalid which no longer accepts an exception as it's second argument, but does take the SQL and binds. Use self.class for the InsertAttributeStripper transaction. ActiveRecord::Base may reference a different database - Currently in springbuk, ActiveRecord::Base goes to the postgres database. self.class will go to the database of whatever parent class (correctly going to snowflake where it's used)
- MOJ-203 Discovered that snowflake returns slightly different types enumerations when you enable utf8.
- changed Integer to bigInteger
- Fix associated record saving
- Moj 450 update with prefetch primary key (#19)
- Do not use bool alias (#21)
- MOJ-366 Two updates for added functionality (#20)
- [COR-3018] Support Rails 7 (#22)
- Moj 574 speedup hr uploads processing (#23)
- [MOJ-578] Use new springbuk odbc-ruby gem. (#24)
- [MOJ-586] Update odbc-ruby gem. (#26)
- Moj 519 plan years controller (#27)
- APP-3029 Fixes prune duplicates. All records with a nil primary key get their primary key replaced by a new object, which will never compare true to anything else. (#28)
